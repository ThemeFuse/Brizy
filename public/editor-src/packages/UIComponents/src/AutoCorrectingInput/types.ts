import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
