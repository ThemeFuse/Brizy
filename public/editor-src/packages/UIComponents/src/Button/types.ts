import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
