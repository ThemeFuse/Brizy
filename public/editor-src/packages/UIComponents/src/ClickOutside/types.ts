import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
