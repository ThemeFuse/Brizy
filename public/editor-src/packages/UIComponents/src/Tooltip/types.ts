import { CSSProperties, MouseEvent, ReactNode, RefObject } from "react";
import { WithClassName } from "../types/attributes";

export type ToolbarItemsInstance = {
  toolbarRef: React.RefObject<HTMLDivElement>;
  toolbarCSSPosition: "fixed" | "absolute";
  toolbarItemIndex: number;
  toolbarItemsLength: number;
};

export type Placement =
  | "auto"
  | "auto-start"
  | "auto-end"
  | "top"
  | "top-start"
  | "top-end"
  | "bottom"
  | "bottom-start"
  | "bottom-end"
  | "right"
  | "right-start"
  | "right-end"
  | "left"
  | "left-start"
  | "left-end";

export type Size = "small" | "medium" | "large" | "xlarge" | "auto";

export interface ContentProps extends WithClassName {
  isOpen?: boolean;
  placement: Placement;
  showArrow?: boolean;
  size?: Size;
  offset?: number;
  toolbar?: ToolbarItemsInstance;
  node?: HTMLElement;
}

export interface ContentState {
  placement: Placement;
  placementStyle: CSSProperties;
  arrowPlacement: "top" | "bottom";
  arrowPlacementStyle: CSSProperties;
}

export type TooltipItemProps = {
  children: ReactNode;
  className?: string;
  onClick?: (e: MouseEvent<HTMLDivElement>) => void;
};

export interface Props
  extends WithClassName,
    Omit<ContentProps, "node" | "isOpen"> {
  overlayClassName?: string;
  showOnClick?: boolean;
  closeDelay?: number;
  overlay?: ReactNode;
  title?: string;
  isInPortal?: boolean;
  portalNode?: HTMLElement;
  clickOutsideExceptions?: string[];
  nodeRef?: RefObject<HTMLElement>;
  onOpen?: VoidFunction;
  onClose?: VoidFunction;
}

export interface State {
  isOpen: boolean;
  isHidden: boolean;
  needClose: boolean;
}
