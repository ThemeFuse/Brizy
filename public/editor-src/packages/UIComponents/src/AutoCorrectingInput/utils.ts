import { clamp } from "../utils/math";

export function formatInputValue(value: number, step: number): string | number {
  const stepDecimalLength = decimalLength(step);

  return stepDecimalLength > 0
    ? value.toFixed(stepDecimalLength)
    : String(value);
}

export function decimalLength(number: number): number {
  const [, decimal] = String(number).split(".");
  return decimal ? decimal.length : 0;
}

export function correctNumber(
  number: number,
  min: number,
  max: number,
  step: number
): number {
  return roundByStep(clamp(number, min, max), step, min);
}

export const preciseAdd = preciseOperation.bind(null, "add");

export const preciseSub = preciseOperation.bind(null, "sub");

function roundByStep(number: number, step: number, offset = 0) {
  // this is done to avoid quirks with floating point operations
  // like 1.1 - 1 = 0.10000000000000009
  const mul = 10 ** decimalLength(step);

  number = number * mul;
  step = step * mul;
  offset = offset * mul;

  const tmp = Math.ceil((number - offset) / step) * step + offset;

  return tmp / mul;
}

function preciseOperation(type: "add" | "sub", a: number, b: number) {
  const dla = decimalLength(a);
  const dlb = decimalLength(b);
  const dlMax = Math.max(dla, dlb);
  const mul = 10 ** dlMax;

  const a_ = a * mul;
  const b_ = b * mul;
  let res;

  switch (type) {
    case "add":
      res = a_ + b_;
      break;
    case "sub":
      res = a_ - b_;
      break;
    default:
      throw new Error(`operation type unknown ${type}`);
  }

  return res / mul;
}
