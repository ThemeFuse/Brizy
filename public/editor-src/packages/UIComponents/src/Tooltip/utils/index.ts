type PositionState = "above" | "below";

const positionState: PositionState = "above";

export const getPosition = (): PositionState => positionState;
