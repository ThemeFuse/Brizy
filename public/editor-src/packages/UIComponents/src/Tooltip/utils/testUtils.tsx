import { ToolbarItemsInstance } from "../types";
import { Placement, Size } from "../types";

export const getToolbar = () =>
  ({
    toolbarRef: { current: document.createElement("div") },
    toolbarCSSPosition: "absolute",
    toolbarItemIndex: 0,
    toolbarItemsLength: 1
  } as unknown as ToolbarItemsInstance);

export const sizes: Size[] = ["auto", "small", "medium", "large", "xlarge"];
export const placements: Placement[] = [
  "auto",
  "auto-start",
  "auto-end",
  "top",
  "top-start",
  "top-end",
  "bottom",
  "bottom-start",
  "bottom-end",
  "right",
  "right-start",
  "right-end",
  "left",
  "left-start",
  "left-end"
];
export const getOffsetPosition = (placement: Omit<Placement, "auto">) => {
  switch (placement) {
    case "right-start":
    case "right-end":
    case "right":
      return "left";
    case "left-start":
    case "left-end":
    case "left":
      return "right";
    case "top-start":
    case "top-end":
    case "top":
      return "bottom";
    case "bottom-start":
    case "bottom-end":
    case "bottom":
      return "top";
  }
};
