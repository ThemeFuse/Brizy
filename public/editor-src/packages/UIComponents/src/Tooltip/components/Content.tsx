import classnames from "classnames";
import React, { CSSProperties, ReactElement } from "react";
import { Popper } from "react-popper";
import { clamp } from "../../utils/math";
import { TOOLBAR_HEIGHT_MAGIC, TOOLBAR_MARGIN } from "../constants";
import { ToolbarItemsInstance } from "../types";
import { ContentProps, ContentState } from "../types";
import { getPosition as getToolbarPosition } from "../utils";

export class TooltipContent extends React.Component<
  ContentProps,
  ContentState
> {
  isRepositioning = false;
  contentRef: React.RefObject<HTMLDivElement> = React.createRef();

  constructor(props: ContentProps) {
    super(props);

    const { placement } = props;
    this.state = {
      placement: placement,
      placementStyle: {},
      arrowPlacement: "top",
      arrowPlacementStyle: {}
    };
  }

  componentDidMount(): void {
    const { isOpen, toolbar } = this.props;

    if (isOpen && toolbar) {
      this.repositionByToolbar(toolbar);
    }
  }

  repositionByToolbar(toolbar: ToolbarItemsInstance): void {
    const { toolbarRef, toolbarCSSPosition, toolbarItemIndex } = toolbar;
    const toolbarNode = toolbarRef.current;
    const toolbarItemNode = toolbarNode?.querySelector(
      `.brz-ed-toolbar__items > .brz-ed-toolbar__item:nth-child(${toolbarItemIndex})`
    );
    const toolbarItemRect = toolbarItemNode?.getBoundingClientRect();

    if (
      !toolbarRef.current ||
      !this.contentRef.current ||
      !toolbarItemRect ||
      !toolbarNode
    ) {
      return;
    }
    const viewportWidth = document.documentElement.clientWidth;
    const viewportHeight = document.documentElement.clientHeight;
    const pageScroll = document.documentElement.scrollTop;
    const pageHasOverflowHidden =
      document.documentElement.classList.contains("brz-ow-hidden");

    const toolbarPosition = getToolbarPosition();

    const toolbarRect = toolbarNode.getBoundingClientRect();

    const tooltipNode = this.contentRef.current;
    const tooltipRect = tooltipNode.getBoundingClientRect();

    // Toolbar

    const canAlignBelow =
      (pageHasOverflowHidden ? 0 : pageScroll) +
        toolbarRect.bottom +
        TOOLBAR_MARGIN +
        Math.max(tooltipRect.height, TOOLBAR_HEIGHT_MAGIC) <=
      (pageHasOverflowHidden
        ? viewportHeight
        : Math.max(viewportHeight, document.body.clientHeight));
    const canAlignAbove =
      (pageHasOverflowHidden ? 0 : pageScroll) +
        toolbarRect.top -
        TOOLBAR_MARGIN -
        tooltipRect.height >=
      0;
    const offsetTop = toolbarCSSPosition === "fixed" ? 0 : pageScroll;
    let placementStyle: CSSProperties = {};

    const getLeft = (): number => {
      const padding = 31;
      const minX = padding;
      const maxX = viewportWidth - tooltipRect.width - padding;
      const x =
        toolbarItemRect.left +
        toolbarItemRect.width / 2 -
        tooltipRect.width / 2;

      return clamp(x, minX, maxX);
    };

    const alignAbove = (): void => {
      const tooltipTop = "unset";
      const tooltipBottom = `calc(100% - ${offsetTop}px - ${toolbarRect.top}px + ${TOOLBAR_MARGIN}px)`;

      placementStyle = {
        position: toolbarCSSPosition,
        top: tooltipTop,
        bottom: tooltipBottom,
        left: getLeft()
      };
    };
    const alignBelow = (): void => {
      const tooltipTop =
        offsetTop + toolbarRect.top + toolbarRect.height + TOOLBAR_MARGIN;

      placementStyle = {
        position: toolbarCSSPosition,
        top: tooltipTop,
        left: getLeft()
      };
    };

    let tooltipAlignment;
    if (toolbarPosition === "above") {
      if (canAlignAbove) {
        tooltipAlignment = "above";
        alignAbove();
      } else {
        tooltipAlignment = "below";
        alignBelow();
      }
    }

    if (toolbarPosition === "below") {
      if (canAlignBelow) {
        tooltipAlignment = "below";
        alignBelow();
      } else {
        tooltipAlignment = "above";
        alignAbove();
      }
    }

    // Arrow

    const arrowPlacement = tooltipAlignment === "above" ? "top" : "bottom";
    const arrowPosition =
      getLeft() === 31 ? toolbarItemRect.width - 21 : toolbarItemRect.width / 2;

    const arrowPlacementStyle = {
      left:
        toolbarItemRect.left +
        arrowPosition -
        (Number(placementStyle.left) || 0)
    };

    this.isRepositioning = true;
    this.setState(
      {
        placementStyle,
        arrowPlacement,
        arrowPlacementStyle
      },
      () => (this.isRepositioning = false)
    );
  }

  renderInToolbar(): ReactElement {
    const {
      className: _className,
      isOpen,
      size,
      showArrow: arrow,
      children
    } = this.props;
    const { placementStyle, arrowPlacement, arrowPlacementStyle } = this.state;

    const className = classnames(
      "brz-ed-animated brz-ed-animated--fadeInUp",
      "brz-ed-tooltip__overlay",
      { [`brz-ed-tooltip--${size}`]: size },
      { "brz-invisible": !isOpen },
      _className
    );
    const arrowClassName = classnames(
      "brz-ed-arrow",
      `brz-ed-arrow--${arrowPlacement}`
    );

    return (
      <div ref={this.contentRef} className={className} style={placementStyle}>
        {arrow && (
          <div className={arrowClassName} style={arrowPlacementStyle} />
        )}
        {children}
      </div>
    );
  }

  renderSimple(): ReactElement {
    const {
      className: _className,
      size,
      showArrow,
      node,
      placement,
      offset,
      children
    } = this.props;
    const className = classnames(
      "brz-ed-animated brz-ed-animated--fadeInUp",
      "brz-ed-tooltip__overlay",
      { [`brz-ed-tooltip--${size}`]: size },
      _className
    );

    return (
      <Popper
        referenceElement={node}
        placement={placement}
        modifiers={[
          {
            name: "offset",
            options: { offset: [0, offset] }
          },
          {
            name: "computeStyles",
            options: { gpuAcceleration: false }
          }
        ]}
      >
        {({ ref, style, arrowProps, placement }): ReactElement => (
          <div ref={ref} className={className} style={style}>
            {showArrow && (
              <div
                className={`brz-ed-arrow brz-ed-arrow--${placement}`}
                ref={arrowProps.ref}
                style={{ ...arrowProps.style, margin: "0" }}
              />
            )}
            {children}
          </div>
        )}
      </Popper>
    );
  }

  render(): ReactElement {
    return this.props.toolbar ? this.renderInToolbar() : this.renderSimple();
  }
}
