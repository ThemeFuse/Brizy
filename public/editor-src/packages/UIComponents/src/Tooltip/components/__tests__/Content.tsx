import { render } from "@testing-library/react";
import React, { PropsWithChildren } from "react";
import { ContentProps, Placement, Size } from "../../types";
import { getToolbar, placements, sizes } from "../../utils/testUtils";
import { TooltipContent } from "../Content";

describe("Test TooltipContent", () => {
  describe("Snapshots Simple render", () => {
    test("Should have children and by default shouldn't have arrow", () => {
      const { container } = render(
        <TooltipContent placement="top">
          <p className="children">children</p>
        </TooltipContent>
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(".brz-ed-tooltip__overlay .children")
      ).toBeInTheDocument();
      expect(container.querySelector(".brz-ed-arrow")).not.toBeInTheDocument();
    });

    test("className should be className", () => {
      const { container } = render(
        <TooltipContent className="className" placement="top">
          Children
        </TooltipContent>
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(container.querySelector(".brz-ed-tooltip__overlay")).toHaveClass(
        "className"
      );
    });

    test("when show Arrow should have children with className:brz-ed-arrow", () => {
      const { container } = render(
        <TooltipContent showArrow={true} placement="top">
          Children
        </TooltipContent>
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(container.querySelector(".brz-ed-arrow")).toBeInTheDocument();
    });

    test.each<Size>(sizes)("Should have size:%s", (size) => {
      const { container } = render(
        <TooltipContent size={size} placement="top">
          children
        </TooltipContent>
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(`.brz-ed-tooltip--${size}`)
      ).toBeInTheDocument();
    });

    test.each<Placement>(placements)(
      "Should have placement:%s",
      (placement) => {
        const { container } = render(
          <TooltipContent placement={placement} showArrow>
            children
          </TooltipContent>
        );

        expect(container.firstChild).toMatchSnapshot();

        expect(
          container.querySelector(`.brz-ed-arrow--${placement}`)
        ).toBeInTheDocument();
      }
    );
  });
  describe("Snapshots render in Toolbar", () => {
    const component = ({
      children,
      ...props
    }: PropsWithChildren<ContentProps>) => (
      <TooltipContent {...props} toolbar={getToolbar()}>
        {children ?? "children"}
      </TooltipContent>
    );
    test("Should have children and by default should be closed", () => {
      const { container } = render(
        component({
          children: <p className="children">children</p>,
          placement: "top"
        })
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(".brz-ed-tooltip__overlay .children")
      ).toBeInTheDocument();
      expect(container.querySelector(".brz-invisible")).toBeInTheDocument();
    });

    test("className should be className", () => {
      const { container } = render(
        component({ className: "className", placement: "top" })
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(container.querySelector(".brz-ed-tooltip__overlay")).toHaveClass(
        "className"
      );
    });

    test("when isn't show Arrow shouldn't have children with className:brz-ed-arrow", () => {
      const { container } = render(
        component({ showArrow: false, placement: "top" })
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(container.querySelector(".brz-ed-arrow")).not.toBeInTheDocument();
    });

    test("when show Arrow should have children with className:brz-ed-arrow", () => {
      const { container } = render(
        component({ showArrow: true, placement: "top" })
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(container.querySelector(".brz-ed-arrow")).toBeInTheDocument();
    });

    test.each<Size>(sizes)("Should have size:%s", (size) => {
      const { container } = render(component({ size, placement: "top" }));

      expect(container.firstChild).toMatchSnapshot();

      expect(
        container.querySelector(`.brz-ed-tooltip--${size}`)
      ).toBeInTheDocument();
    });

    test("when isOpen shouldn't have className:brz-invisible", () => {
      const { container } = render(
        component({ isOpen: true, placement: "top" })
      );

      expect(container.firstChild).toMatchSnapshot();

      expect(container.querySelector(".brz-invisible")).not.toBeInTheDocument();
    });
  });
});
