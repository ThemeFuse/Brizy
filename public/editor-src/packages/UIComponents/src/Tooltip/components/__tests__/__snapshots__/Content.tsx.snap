// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TooltipContent Snapshots Simple render Should have children and by default shouldn't have arrow 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <p
    class="children"
  >
    children
  </p>
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:auto 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--auto"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:auto-end 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--auto-end"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:auto-start 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--auto-start"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:bottom 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--bottom"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:bottom-end 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--bottom-end"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:bottom-start 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--bottom-start"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:left 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--left"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:left-end 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--left-end"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:left-start 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--left-start"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:right 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--right"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:right-end 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--right-end"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:right-start 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--right-start"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:top 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--top"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:top-end 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--top-end"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have placement:top-start 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--top-start"
    style="position: absolute; margin: 0px;"
  />
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have size:auto 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--auto"
  style="position: absolute; left: 0px; top: 0px;"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have size:large 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--large"
  style="position: absolute; left: 0px; top: 0px;"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have size:medium 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--medium"
  style="position: absolute; left: 0px; top: 0px;"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have size:small 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--small"
  style="position: absolute; left: 0px; top: 0px;"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render Should have size:xlarge 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--xlarge"
  style="position: absolute; left: 0px; top: 0px;"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render className should be className 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay className"
  style="position: absolute; left: 0px; top: 0px;"
>
  Children
</div>
`;

exports[`Test TooltipContent Snapshots Simple render when show Arrow should have children with className:brz-ed-arrow 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
  style="position: absolute; left: 0px; top: 0px;"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--top"
    style="position: absolute; margin: 0px;"
  />
  Children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar Should have children and by default should be closed 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-invisible"
>
  <p
    class="children"
  >
    children
  </p>
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar Should have size:auto 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--auto brz-invisible"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar Should have size:large 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--large brz-invisible"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar Should have size:medium 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--medium brz-invisible"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar Should have size:small 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--small brz-invisible"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar Should have size:xlarge 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--xlarge brz-invisible"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar className should be className 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-invisible className"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar when isOpen shouldn't have className:brz-invisible 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar when isn't show Arrow shouldn't have children with className:brz-ed-arrow 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-invisible"
>
  children
</div>
`;

exports[`Test TooltipContent Snapshots render in Toolbar when show Arrow should have children with className:brz-ed-arrow 1`] = `
<div
  class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-invisible"
>
  <div
    class="brz-ed-arrow brz-ed-arrow--top"
  />
  children
</div>
`;
